### PR TITLE
feat(framework) Update message handler methods to raise `AbortRunException` if called but run is not running

### PR DESCRIPTION
**Pull request by chongshenng**
Merge after:
* #4693

### CODE REVIEW

File: src/py/flwr/server/superlink/fleet/grpc_rere/server_interceptor_test.py [hunk 1]
```diff
@@ -275,8 +276,14 @@ def test_successful_push_task_res_with_metadata(self) -> None:
...
             ping_interval=30, public_key=public_key_to_bytes(self._node_public_key)
         )
+        run_id = self.state.create_run("", "", "", {}, ConfigsRecord())
+        # Transition status to running. PushTaskRes is only allowed in running status.
+        _ = self.state.update_run_status(run_id, RunStatus(Status.STARTING, "", ""))
+        _ = self.state.update_run_status(run_id, RunStatus(Status.RUNNING, "", ""))
         request = PushTaskResRequest(
-            task_res_list=[TaskRes(task=Task(producer=Node(node_id=node_id)))]
+            task_res_list=[
+                TaskRes(task=Task(producer=Node(node_id=node_id)), run_id=run_id)
```
Comment (jafermarq) [inline src/py/flwr/server/superlink/fleet/grpc_rere/server_interceptor_test.py:285]: I note we create a `TaskRes` in another test below this one. Should we pass the `run_id` in there too for consistency?

Comment (chongshenng) [reply src/py/flwr/server/superlink/fleet/grpc_rere/server_interceptor_test.py:285]: Agreed, done.

**Review by jafermarq — DISMISSED** [review]

File: src/py/flwr/server/superlink/fleet/message_handler/message_handler.py [hunk 2]
```diff
@@ -121,6 +132,15 @@ def get_run(
     if run is None:
         return GetRunResponse()
 
+    # Abort if the run is not running
+    abort_msg = check_abort(
+        request.run_id,
+        [Status.PENDING, Status.STARTING, Status.FINISHED],
```
Comment (danieljanes) [inline src/py/flwr/server/superlink/fleet/message_handler/message_handler.py:138]: Shouldn't the SuperNode be able to get the run even if it's finished?

File: src/py/flwr/common/typing.py [hunk 3]
```diff
@@ -258,3 +258,11 @@ class Fab:
 
 class RunNotRunningException(BaseException):
     """Raised when a run is not running."""
+
+
+class AbortRunException(BaseException):
```
Comment (danieljanes) [inline src/py/flwr/common/typing.py:263]: Should we name this `InvalidRunStatus` or something along those lines? `AbortRun` sounds like the run needs to be aborted, which isn't the case.

File: src/py/flwr/server/superlink/fleet/message_handler/message_handler.py (hunk 2 above)
Comment (chongshenng) [reply src/py/flwr/server/superlink/fleet/message_handler/message_handler.py:138]: Changed to `InvalidRunStatusException` to maintain consistent  `*Exception` naming.

File: src/py/flwr/common/typing.py (hunk 3 above)
Comment (chongshenng) [reply src/py/flwr/common/typing.py]: Changed to InvalidRunStatusException to maintain consistent *Exception naming.

File: src/py/flwr/server/superlink/fleet/message_handler/message_handler.py (hunk 2 above)
Comment (chongshenng) [reply src/py/flwr/server/superlink/fleet/message_handler/message_handler.py:138]: IMO, as far as the workflow goes now, the SuperNode should only be allowed (and able) to pull a Run that is running. At least this is what makes most sense to me.

**Review by danieljanes — APPROVED** [review]

**Pull request merged**